### PR TITLE
DOC: add warning to append about inefficiency

### LINKED ISSUE #16418: DOC: add warning to append about inefficiency
In many cases, using `append` to add rows is not the best way to go, as this is very inefficient (certainly if you do this in a for loop, each time taking a copy)

If in a situation where you have this pattern of appending in a for loop, you typically want to append to a list and then concatenate all at once using `concat`.

Would by good to add a note about this in the `append` docstring.

### PR DESCRIPTION
**Pull request by shanral**
- [ ] closes #16418 
 - [x] adds clarifying description and example for proper use of append
 - [x] passes git diff upstream/master -u -- "*.py" | flake8 --diff

### CODE REVIEW

File: doc/make.py [hunk 1]
```diff
@@ -150,10 +150,13 @@ def _remove_notebooks():
             print("Warning: Pandoc is not installed. Skipping notebooks.")
             _remove_notebooks()
 
-    yield
-    for nb, content in contents.items():
-        with open(nb, 'wt') as f:
-            f.write(content)
+    try:
```
Comment (gfyoung) [inline doc/make.py]: Why this `try`-`except` here?

File: pandas/core/series.py [hunk 2]
```diff
@@ -1522,6 +1522,17 @@ def append(self, to_append, ignore_index=False, verify_integrity=False):
         verify_integrity : boolean, default False
             If True, raise Exception on creating index with duplicates
 
+        Notes
+        -----
+        Iteratively appending to a series can be more computationally intense
+        than a single concatenate. A better solution is to append values to a
+        list then concatenate the list with the original series all at once.
```
Comment (gfyoung) [inline pandas/core/series.py]: series --> Series
intense --> intensive
"list then" --> "list and then"

File: pandas/core/frame.py [hunk 3]
```diff
@@ -4618,6 +4618,11 @@ def append(self, other, ignore_index=False, verify_integrity=False):
         the DataFrame's index, the order of the columns in the resulting
         DataFrame will be unchanged.
 
+        Iteratively appending rows to a Dataframe can be more computationally
+        intense than a single concatenate. A better solution is to append those
+        rows to a list then concatenate the list with the original Dataframe
+        all at once.
```
Comment (gfyoung) [inline pandas/core/frame.py]: intense --> intensive
"list then" --> "list and then"

Comment (jreback) [inline pandas/core/frame.py]: can you show some pseudo code

File: pandas/core/frame.py [hunk 4]
```diff
@@ -4653,6 +4653,32 @@ def append(self, other, ignore_index=False, verify_integrity=False):
         2  5  6
         3  7  8
 
+        The following, while not a recommended method for generating a
+        DataFrame, illustrates how to efficiently generate a DataFrame from
```
Comment (gfyoung) [inline pandas/core/frame.py]: You can't call this "efficient" since you just said it wasn't efficient.

File: pandas/core/frame.py [hunk 5]
```diff
@@ -4653,6 +4653,32 @@ def append(self, other, ignore_index=False, verify_integrity=False):
...
+        multiple data sources.
+
+        Less efficient:
+        >>> df = pd.DataFrame(columns=['A'])
```
Comment (gfyoung) [inline pandas/core/frame.py]: Add a newline between "Less efficient" and ">>> df = ..."

File: pandas/core/series.py [hunk 6]
```diff
@@ -1524,9 +1524,10 @@ def append(self, to_append, ignore_index=False, verify_integrity=False):
 
         Notes
         -----
-        Iteratively appending to a series can be more computationally intense
+        Iteratively appending to a Series can be more computationally intensive
         than a single concatenate. A better solution is to append values to a
-        list then concatenate the list with the original series all at once.
+        list and then concatenate the list with the original series all at
```
Comment (gfyoung) [inline pandas/core/series.py]: series --> Series

**Pull request closed without merging**